src/servers/http_server_v2.cc

```diff
@@ -1177,28 +1217,110 @@ HTTPAPIServerV2::EVBufferToInput(
       // FIXMEV2 handle non-raw content types
       char* base;
       size_t byte_size;
-      RETURN_IF_ERR(ReadDataFromJson(request_input, &base, &byte_size));
+      rapidjson::Value::ConstMemberIterator itr =
+          request_input.FindMember("parameters");
+      if (itr != request_input.MemberEnd()) {
+        if (header_length == 0) {
```

Comment:
Why is this check here?

Reply:
Removed

Reply:
The check is done only when one of the inputs have a member "parameters" i.e. the case when binary data is used.

Reply:
Parameters are not just to indicate binary data, they are/will-be used for other purposes.
Can you write a utility routine like bool IsBinaryInputData(..) that returns true/false and also set parameters for the offset/size. Perhaps having offset just makes the implementation harder? We can just have 'size' and define that the binary data is in same order as inputs in the json. In fact I prefer that since it should be more performant.

Reply:
I agree. Having just size and enforcing the order to be the same would be more performant

Reply:
ok, I will update the extension doc to not include the offset.